corehq/apps/app_manager/models.py

```diff
@@ -4655,18 +4655,13 @@ def get_odk_qr_code(self, with_media=False, build_profile_id=None, download_targ
                 kwargs.append('download_target_version=true')
             url += '?' + '&'.join(kwargs)
 
-            code.add_data(url)
-
-            # "Level L" error correction with a 0 pixel margin
-            code.set_ec('L', 0)
-            f, fname = tempfile.mkstemp()
-            code.download(fname)
-            os.close(f)
-            with open(fname, "rb") as f:
-                png_data = f.read()
-                self.lazy_put_attachment(png_data, filename,
-                                         content_type="image/png")
-            return png_data
+            image = qrcode.make(url)
+            output = BytesIO()
+            image.save(output, "PNG")
+            qr_content = output.getvalue()
```

Comment:
Can these 4 lines be put in a method and used at both places?

Reply:
Possibly but wanted to keep this small. Since they're in different apps would probably want to put them somewhere in a separate util app